dromedary.py

```diff
@@ -163,6 +180,166 @@ def print_output(out: subprocess.CompletedProcess) -> None:
     print(std_out + err_out)
 
 
+###############################################################################
+def run_cmd_output(cmd_args: List[Any], cmd_env: Optional[Dict[str, str]]) -> str:
+    """Run the given command in a shell.
+
+    Prints the output of the process in "real time".
+
+    Args:
+        cmd_args (List[Any]): The command and it's arguments to run.
+        cmd_env (Optional[Dict[str, str]]): The environment to run the command
+                                            in.
+
+    Returns:
+        str: The accumulated stdout and stderr output of the process.
+    """
+    std_out: str = ""
+    with subprocess.Popen(
+        " ".join(map(lambda e: f"'{str(e)}'", cmd_args)),
+        shell=True,
+        env=cmd_env,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    ) as p:  # nosec
+        while p.poll() is None:
+            std_str = p.stdout.read1().decode("utf-8")
+            std_out += std_str
+            print(std_str, end="", flush=True, file=sys.stdout)
+
+        if p.poll() != 0:
+            print(f"Error: command '{p.args}' returned '{p.poll()}'")
+            return error_exit(6)
+
+    return std_out
+
+
+class SwitchConfig(NamedTuple):
+    """Holds valid configuration of a switch to create."""
+    name: str
+    compiler: str
+    packages: List[str]
+
+
+###############################################################################
+def generate_switch(config_file: str) -> Tuple[str, List[str]]:
+    """Generate an Opam switch from the configuration in the given JSON file
+    `config_file`.
+
+    Args:
+        config_file (str): The path to the JSON config file to use.
+
+    Returns:
+        SwitchConfig: A `Tuple` containing the name of the generated Opam
+                      switch and the list of packages to install.
+    """
+    print(f"Using config file '{config_file}'")
+    switch_config = read_json(config_file)
+
+    valid_config = validate_config(switch_config, json_path=config_file)
+    print(
+        f"Generating Opam switch '{valid_config.name}' using compiler '{valid_config.compiler}'"
+    )
+
+    switch_name_rex = r"^\s*#\s*Run\s*eval\s*\$\(opam\s*env\s*--switch=(?P<name>[^)]+)\)\s*to\s*update"
+    create_args = OPAM_SWITCH_CREATE_CMD
+    create_args.extend([valid_config.name, valid_config.compiler])
+    std = run_cmd_output(create_args, cmd_env=None)
+    match = re.search(switch_name_rex, std, re.UNICODE | re.DOTALL | re.MULTILINE)
+    if match is None:
+        print(
+            f"Error: could not successfully parse the output of 'opam switch create':\n{std}\nUsing regex {switch_name_rex}",
```

Comment:
Feedback from @grievejia  and I fully agree (sorry I missed it):

Potentially silly question: what's the point of parsing the output of opam switch create to extract the switch name? Don't we know the name of the switch (valid_config.name) already?
This feels rather fragile as it relies on opam keeping its human-oriented output format stable, which imo is not very likely.

Reply:
As Opam's handling of switch names (which can or must be "compiler names") and the differentiation between paths and "real" names is complex, I do not trust Opam to use the same "name" as it has been given (it happened once to me that the name I passed had not been the same as the one Opam actually generated). 
But again thinking about it, it is better to try and use the "wrong" name and fail when calling `opam env`, than trying to go on using the "actual" switch name. As I actually never liked that part either, let's remove it.